src/HeaderBar.vala

```diff
@@ -171,8 +213,22 @@ public class Mail.HeaderBar : Hdy.HeaderBar {
         });
     }
 
+    private void on_filter_button_changed () {
+        unowned var style_context = filter_button.image.get_style_context ();
+        var has_accent_class = style_context.has_class (Granite.STYLE_CLASS_ACCENT);
+
+        if (hide_read || hide_unstarred) {
+            if (!has_accent_class) {
+                style_context.add_class (Granite.STYLE_CLASS_ACCENT);
+            }
+
+        } else if (has_accent_class) {
+            style_context.remove_class (Granite.STYLE_CLASS_ACCENT);
+        }
+    }
+
     public void set_paned_positions (int start_position, int end_position, bool start_changed = true) {
-        search_entry.width_request = end_position - start_position + 1;
+        search_entry.width_request = end_position - start_position - filter_button.get_allocated_width () - 6;
```

Comment:
I guess this should use the end margin of the filter button so we know where this number comes from
```suggestion
        search_entry.width_request = end_position - start_position - filter_button.get_allocated_width () - filter_button.margin_end;
```